src/D2L.CodeStyle.Analyzers/RpcDependencies/RpcDependencyAnalyzer.cs

```diff
@@ -69,58 +84,108 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
 			INamedTypeSymbol rpcAttributeType,
 			INamedTypeSymbol rpcContextType,
 			INamedTypeSymbol rpcPostContextType,
-			INamedTypeSymbol rpcPostContextBaseType
+			INamedTypeSymbol rpcPostContextBaseType,
+			INamedTypeSymbol dependencyAttributeType
 		) {
 			var method = context.Node as MethodDeclarationSyntax;
 
 			if ( method == null ) {
 				return;
 			}
 
+			bool keepGoing = CheckThatFirstArgumentIsIRpcContext(
+				context,
+				method,
+				rpcAttributeType: rpcAttributeType,
+				rpcContextType: rpcContextType,
+				rpcPostContextType: rpcPostContextType,
+				rpcPostContextBaseType: rpcPostContextBaseType
+			);
+
+			if ( !keepGoing ) {
+				return;
+			}
+
+			CheckThatDependencyArgumentsAreSortedCorrectly(
+				context,
+				method.ParameterList.Parameters,
+				dependencyAttributeType
+			);
+
+			// other things to check:
+			// - appropriate use of [Dependency]
+		}
+
+		private static bool CheckThatFirstArgumentIsIRpcContext(
+			SyntaxNodeAnalysisContext context,
+			MethodDeclarationSyntax method,
+			INamedTypeSymbol rpcAttributeType,
+			INamedTypeSymbol rpcContextType,
+			INamedTypeSymbol rpcPostContextType,
+			INamedTypeSymbol rpcPostContextBaseType
+		) {
 			bool isRpc = method
 				.AttributeLists
 				.SelectMany( al => al.Attributes )
-				.Any( attr => IsRpcAttribute( rpcAttributeType, attr, context.SemanticModel ) );
+				.Any( attr => IsAttribute( rpcAttributeType, attr, context.SemanticModel ) );
 
 			if( !isRpc ) {
-				return;
+				return false;

```

Comment:
I feel like this should be in `CheckThatMethodIsRpcMethod`, that is called before all of the other `CheckThat*` methods.